wasm/jit/jit_amd64.go

```diff
@@ -1096,6 +1098,90 @@ func (b *amd64Builder) handleLe(o *wazeroir.OperationLe) error {
 	return nil
 }
 
+func (b *amd64Builder) handleLoad(o *wazeroir.OperationLoad) error {
+	base := b.locationStack.pop()
+	if base.onStack() {
+		if err := b.moveStackToRegisterWithAllocation(generalPurposeRegisterTypeInt, base); err != nil {
+			return err
+		}
+	} else if base.onConditionalRegister() {
+		if err := b.moveConditionalToGPRegister(base); err != nil {
+			return err
+		}
+	}
+
+	// At this point, base's value is on the integer general purpose reg.
+	// We reuse the register below, so we alias it here for readability.
+	reg := base.register
+
+	// Then we have to calculate the offset on the memory region.
+	addOffsetToBase := b.newProg()
+	addOffsetToBase.As = x86.AADDL // 32-bit!
+	addOffsetToBase.To.Type = obj.TYPE_REG
+	addOffsetToBase.To.Reg = reg
+	addOffsetToBase.From.Type = obj.TYPE_CONST
+	addOffsetToBase.From.Offset = int64(o.Arg.Offest)
+	b.addInstruction(addOffsetToBase)
+
+	// TODO: Emit instructions here to check memory out of bounds as
+	// potentially it would be an security risk.
+
+	var (
+		isIntType bool
+		movInst   obj.As
+	)
+	switch o.Type {
+	case wazeroir.SignLessTypeI32:
```

Comment:
so assuming signless is not the same as unsigned, would it be better to use wasm conventions and say `TypeI32` (instead of `TypeU32`)? https://www.w3.org/TR/wasm-core-1/#integers%E2%91%A0

Reply:
> so assuming signless is not the same as unsigned

well, signless is the same as unsigned, so I think it is fine. (Maybe we should rethink about the name "Signless")

Reply:
hmm unsigned is far more common language than signless, and it is used in the wasm spec.. if they are the same, how about preferring unsigned then?

Reply:
will raise a PR to rename shortly